### PR TITLE
Adding fp32 horizontal diffusion design

### PR DESCRIPTION
**Pull request by singagan**

### CODE REVIEW

**Review by AndraBisca — CHANGES_REQUESTED** [review]
The datasets are quite big, I don't think we want to upload them, just the source files. Also, please apply the clang format patch before any merge.

**Comment by singagan** [comment]
Removed datasets for now.

**Comment by singagan** [comment]
Added the mlir-aie open source license

**Review by denolf — CHANGES_REQUESTED** [review]
The reference design folder should not contain every variant of the weather stencils. Only the variants contributing to the final reference design should be included. See PR #305.

**Comment by singagan** [comment]
Adding only the variants contributing to the final design

**Review by AndraBisca — APPROVED** [review]
Thank you for making the changes!

File: reference_designs/horizontal_diffusion/HDIFF_dual_AIE_objectFIFO/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+f32: f32_chess f32.elf
+
+build: 
+	xchessmk test.prx
+sim: 
+	xca_udm_dbg -P /mnt/scratch/tools_tmp/Xilinx/Vitis/2021.2/aietools/data/cervino/lib/ -t sim.tcl
+
+
+i32_chess: 
+	/mnt/scratch/tools_tmp/Xilinx/Vitis/2021.2/aietools/bin/xchesscc -p me -P /mnt/scratch/tools_tmp/Xilinx/Vitis/2021.2/aietools/data/cervino/lib/ -L /mnt/scratch/tools_tmp/Xilinx/Vitis/2021.2/aietools/data/cervino/lib/ -c ./hdiff_lap.cc ./hdiff_flux.cc
```
Comment (denolf) [inline reference_designs/horizontal_diffusion/HDIFF_dual_AIE_objectFIFO/Makefile:21]: Remove hard coded paths to your environment. Use the makefile variable (Line 7 and 8 instead). Apply this to all Makefiles. Thanks

File: reference_designs/horizontal_diffusion/HDIFF_dual_AIE_objectFIFO/hdiff_flux_fp32.cc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    // v8int32 * restrict  ptr_in = (v8int32 *) in;
+    // v8int32 * restrict  ptr_in = (v8int32 *) in;
+    v8float * restrict  ptr_forward = (v8float *) flux_forward1;
+    v8float * ptr_out = (v8float *) out;
+    
+    v8float * restrict row1_ptr=(v8float *)row1;
+    v8float * restrict row2_ptr=(v8float *)row2;
+    v8float * restrict row3_ptr=(v8float *)row3;
+
+    // v8int32 * restrict r1=(v8int32 *)row1_ptr;
```
Comment (denolf) [inline reference_designs/horizontal_diffusion/HDIFF_dual_AIE_objectFIFO/hdiff_flux_fp32.cc:33]: remove old code instead of leaving it in comments

**Pull request closed without merging**